### ISSUE TITLE
Copy and paste from the Windows Terminal doesn't include line breaks

### ISSUE THREAD

**Issue by conioh** [microsoft/terminal#1073]
# Environment

```none
Windows build number: 10.0.18362.116
Windows Terminal version (if applicable): 71e19cd + changing toolset to v142 and SDK version
```

# Steps to reproduce

1. Clone, build, package, install and launch _Windows Terminal (DevBuild)_.
2. Execute `docker run --rm -it mcr.microsoft.com/windows/nanoserver:1903`
3. Select multiple lines and right click.
4. Paste in any text editor. I used VSCode.

# Expected behavior

I expected to see the same thing I saw in the terminal - specifically multiple lines.

# Actual behavior

One line with lots of space.

# Notes

This is different from #65 since it was about conhost and I'm talking about the new terminal, and I didn't have Shift pressed.

**Comment by zadjii-msft** [comment]
Huh, I'm not seeing this - @carlos-zamora thoughts?

![image](https://user-images.githubusercontent.com/18356694/58705245-3e872500-8374-11e9-8597-0316cf3517a9.png)

**Comment by d-bingham** [comment]
@zadjii-msft There's some terminal state change that causes it.  Repro steps:

1) Open up a bash.exe session
2) ls -l to generate some multiline text
3) Select and copy it -- works as expected
4) edit a file with vi
5) Select and copy text -- now the endlines are missing

![image](https://user-images.githubusercontent.com/26487560/58743194-a2e8c980-83f2-11e9-9932-3c479fa4ee55.png)

I've been well aware of this working on the clipboard stuff, but haven't tracked down the bug yet.  The bug probably manifests in TextBuffer::GetTextForClipboard, but I haven't had the time to debug it yet.

**Comment by DHowett-MSFT** [comment]
Closing in favor of #1091.

**Comment by DHowett-MSFT** [comment]
It turns out that this is a different issue.

**Comment by conioh** [comment]
I edited the issue to add correct instruction to reproduce.
@d-bingham @zadjii-msft, it happens with docker and not only with WSL.
Also happens if I open the console with `hcsdiag console id` instead of using `docker run`.

**Comment by ljani** [comment]
I'm seeing this as well. My shell is Powershell Core 6.2.0 and everything I try to copy has this problem. It seems that the copy functionality copies the whole buffer. Ie, if I'm trying to copy this:

```plain
get-content myfile.txt
Hello
World

123456789012345678901234567890
```

It copies this, where `^` denotes a space and my terminal is 30 chars wide:

```plain
get-content^myfile.txt^^^^^^^^
Hello^^^^^^^^^^^^^^^^^^^^^^^^^
World^^^^^^^^^^^^^^^^^^^^^^^^^
^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
123456789012345678901234567890
```

**Comment by andrp92** [comment]
I'm having same issue. when entering a copied command like: 

>  apt-get install -y make build-essential checkinstall libssl-dev libbz2-dev libzip-dev \
    libreadline-gplv2-dev libncursesw5-dev libsqlite3-dev tk-dev libgdbm-dev libc6-dev libffi-dev \
    wget curl libldap2-dev zlib1g-dev lib32z1-dev uuid-dev

It only runs the very first line... when using WSL directly, works as expected.

**Comment by mikepurvis** [comment]
I also have this issue when copying text out of an SSH session to a Linux host.

Currently working around it by having this script on my desktop and giving it a double click when I have something I need to paste out of Terminal:

```
from pyperclip import copy, paste                                                                                                                                                                                                                from re import sub                                                                                                                                                                                                                               copy(sub(' {10,}', '\r\n', paste()))
```

**Comment by hanubeki** [comment]
I think I found an easier reproduce steps.

1. Open cmd or PowerShell
2. `cls` to clear the screen
3. Run some command that writes output such as `dir`
4. Copy output text
5. Paste to somewhere (e.g. notepad)
6. Pasted text doesn't include line breaks

**Comment by erwa** [comment]
I'm also facing this issue when running WSL Ubuntu distribution. If I open a WSL tab in Windows Terminal and run something like `ls -1`, then copy some lines using my mouse (hold left-click, drag, then right-click), the copied text does not include any newlines.

Note that if I directly open WSL Ubuntu, copying works fine (the newlines are included).

**Comment by apenney** [comment]
I've tried all the workarounds I've seen listed, from ctrl-c, shift-ctrl-c, right clicks, shift right clicks, etc, but I'm having an absolute nightmare with pasting out of WSL.

It's definitely not handling newlines right, and sometimes when I paste stuff into neovim I get an extra line between every single line.  It's very weird, it's almost enough to stop me using the terminal at this point. :(

**Comment by BartTaylor** [comment]
> I've tried all the workarounds I've seen listed, from ctrl-c, shift-ctrl-c, right clicks, shift right clicks, etc, but I'm having an absolute nightmare with pasting out of WSL.
> 
> It's definitely not handling newlines right, and sometimes when I paste stuff into neovim I get an extra line between every single line. It's very weird, it's almost enough to stop me using the terminal at this point. :(

Me too. This is the one feature (copying from terminal) that I rely on heavily, so I can't really use this until it's fixed. :|

**Comment by hdformat** [comment]
I'm facing same issue, I've tried following for reproduce steps

original text
----
```
(venv) PS hdformat> Get-Content .\test.txt
[2019-08-27 10:42:25,084: INFO/ForkPoolWorker-1] scheduler.push_new.send_push_on_time[c6a36103-80f9-44e8-8ddd-f406b563cded]: waiting countdown=9:48:00
[2019-08-27 10:43:25,051: INFO/ForkPoolWorker-2] ドキドキ
[2019-08-27 10:40:24,915: INFO/ForkPoolWorker-2] 두근두근
```

Windows Console (120 columns, powershell core 6.2.1)

right click
---
(venv) PS hdformat> Get-Content .\test.txt
[2019-08-27 10:42:25,084: INFO/ForkPoolWorker-1] scheduler.push_new.send_push_on_time[c6a36103-80f9-44e8-8ddd-f406b563cded]: waiting countdown=9:48:00
[2019-08-27 10:43:25,051: INFO/ForkPoolWorker-2] ドキドキ
[2019-08-27 10:40:24,915: INFO/ForkPoolWorker-2] 두근두근

Windows Terminal 0.4 (120 columns, powershell core 6.2.1)
----

right click
```
---------1---------2---------3---------4---------5---------6---------7---------8---------9---------0---------1---------2---------3---------4---------5---------6---------7---------8---------9---------0---------1---------2---------3---------4---------5---------6---------7---------8---------9---------0
(venv) PS hdformat> Get-Content .\test.txt
[2019-08-27 10:42:25,084: INFO/ForkPoolWorker-1] scheduler.push_new.send_push_on_time[c6a36103-80f9-44e8-8ddd-f406b563cded]: waiting countdown=9:48:00                                                                                          [2019-08-27 10:43:25,051: INFO/ForkPoolWorker-2] ドキドキ
[2019-08-27 10:40:24,915: INFO/ForkPoolWorker-2] 두근두근
```

alt + c (binding `copyTextWithoutNewlines`) & right click
```
---------1---------2---------3---------4---------5---------6---------7---------8---------9---------0---------1---------2---------3---------4---------5---------6---------7---------8---------9---------0---------1---------2---------3---------4---------5---------6---------7---------8---------9---------0
(venv) PS hdformat> Get-Content .\test.txt                                                                              [2019-08-27 10:42:25,084: INFO/ForkPoolWorker-1] scheduler.push_new.send_push_on_time[c6a36103-80f9-44e8-8ddd-f406b563cded]: waiting countdown=9:48:00                                                                                          [2019-08-27 10:43:25,051: INFO/ForkPoolWorker-2] ドキドキ                                                               [2019-08-27 10:40:24,915: INFO/ForkPoolWorker-2] 두근두근                                                               (venv) PS hdformat>   
```

shift + right click & right clcik
```
---------1---------2---------3---------4---------5---------6---------7---------8---------9---------0---------1---------2---------3---------4---------5---------6---------7---------8---------9---------0---------1---------2---------3---------4---------5---------6---------7---------8---------9---------0
(venv) PS hdformat> Get-Content .\test.txt                                                                              [2019-08-27 10:42:25,084: INFO/ForkPoolWorker-1] scheduler.push_new.send_push_on_time[c6a36103-80f9-44e8-8ddd-f406b563cded]: waiting countdown=9:48:00                                                                                          [2019-08-27 10:43:25,051: INFO/ForkPoolWorker-2] ドキドキ                                                               [2019-08-27 10:40:24,915: INFO/ForkPoolWorker-2] 두근두근                                                               (
```

**Comment by illera88** [comment]
Hi guys!

I'm facing the same problem. Is there any ETA to fix this? As someone mentioned this bug makes very difficult to use the Terminal since every developer relies heavily on copy and pasting.

Thanks for your effort!

**Comment by awarmfastbear** [comment]
I am also still seeing this, pretty frustrating, haven't found any workarounds yet, trying to figure out if my bash / zsh configuration is contributing to this. Right click copy, copy on select, ctrl+shift+c, all copy a ton of whitespace instead of linebreaks.

Windows Terminal Preview - Version: 0.4.2382.0

**Comment by EdLy3** [comment]
Same problem in a DOS window.  Totally unusable.  I copy out of my text editor in a DOS window and paste into a text email within Thunderbird and I have garbage.  It worked up until I upgraded to 1903.  This needs to be fixed.  Or add an option.  This might be useful when pasting into a word processor where you want to reformat but not the way I use it.  I have been able to use this capability for many years and now it is broken.

**Comment by zadjii-msft** [comment]
@EdLy3 Could you file a separate issue to track that regression? This issue is specifically for the missing line breaks in copied text from the **Windows Terminal**, what it sounds like you're seeing is a regression in `conhost.exe`, the legacy console for cmd.exe, powershell.exe, wsl.exe, etc. Make sure to include what commandline application you're copying from. Thanks!

**Comment by mariomarkus** [comment]
I ssh into a linux server, and need to copy-paste some text both ways. 
When copying from notepad and pasting into the terminal window, there are empty lines between lines. 
When copying from terminal window and pasting into Notepad++, the newlines are gone, and there are spaces for all the selected area - that means the size of the pasted text is dependent on the terminal window size at the time of copying.

Just as a remark: when I use a Powershell window for ssh, copypaste works great, without any of these issues.

**Comment by Deathshiver** [comment]
I just wanted to add I'm also seeing this issue with Windows Terminal 0.4.2382.0. I can confirm previous comments are correct. Issue seems related to state and affects tabs individually. I can reproduce quickly using both cmd and PowerShell using the `cls` method posted above, and my WSL immediately falls into the bad state on start (I assume based on my shell setup). After the tab gets into a bad state, it appears to not recover. Only way I have been able to reliably fix the issue is closing and re-opening the tab.

This isn't a set of reproduction steps, but on WSL I'm running zsh with "oh my zsh", "af-magic" theme, and "zsh-syntax-highlighting" plugin. Copy+paste is immediately broken on any command (no newlines, lots of whitespace). Perhaps that helps someone create a faster test.

**Comment by mikkushukla** [comment]
Any progress on this issue ?

**Comment by DHowett-MSFT** [comment]
> Turns out, the clipboard code is correct. ConHost uses the same clipboard code. The issue arises with the `wrap` property on the `CharRows`. Windows Terminal does not set them properly. Significantly more info below.
> 
> In the example above, say the output is displayed as follows (line numbers on left):
> 
> ```
> 0 c:\project-system>set path                                                                                                                                                                                    
> 1 Path=C:\Program Files (x86)\Microsoft Visual Studio\2019\master\Common7\IDE\CommonExtensions\Microsoft\TestWindow;C:\Users\dav
> 2 kean\AppData\Local\Microsoft\WindowsApps;C:\Users\davkean\.dotnet\tools                                                                   
> 3 PATHEXT=.COM;.EXE;.BAT;.CMD;.VBS;.VBE;.JS;.JSE;.WSF;.WSH;.MSC 
> ```
> 
> This means that the wrap is supposed to be...
> 
> | Line # |      wrap? (ConHost)      |     wrap? (Windows Terminal)      |
> |-- |-- |-- |
> | 0 |  false |  false |
> | 1 |  true | true |
> | 2 |  false | **true** |
> | 3 |  false | **true** |
> 
> @miniksa and I tried doing a proper fix in branch `dev/cazamor/bugfix-copy-newline`, but that caused more issues to arise. Specifically, take a look at line 2. In the VT Renderer, we called `_EraseCharacter()` to put spaces until the end of the line. We noticed that that would trigger the text buffer to set that line as "wrapping". The branch changes it to instead call `_EraseLine()`, which should be more efficient too.
> 
> We also updated the `TerminalApi` for `Terminal Core` to `writeLine()` instead of just `Write()` in `Terminal::EraseInLine()`.
> 
> The result was the following:
> 
> | Line # |      wrap? (dev/cazamor/bugfix-copy-newline)      |
> |-- |-- |
> | 0 |  false |
> | 1 |  **false** |
> | 2 |  false |
> | 3 |  false |
> 
> @miniksa was saying that this is a combined problem between the TerminalApi and the VTRenderer. May also be related to #780.
> 
> @zadjii-msft @miniksa Thoughts?
> Removing the "Help-Wanted" tag because this is out of that scope now.
> 
> EDIT: I just tried reverting the change to the TerminalApi and this particular instance is fixed. Is that correct?


Turns out that duplicate #2275 was the same as this, and a bunch of info got put in there.

**Comment by awarmfastbear** [comment]
@DHowett Wait, the duplicate #2275, you referenced is closed which was opened after this one, and the code change / PR for #2275 was later deemed unfit and closed out...I don't see any other avenues on how this is going to be resolved. Can you please clarify if I missed another PR or open issue?

The PR in question from #2275 https://github.com/microsoft/terminal/pull/2755

**Comment by carlos-zamora** [comment]
> @DHowett Wait, the duplicate #2275, you referenced is closed which was opened after this one, and the code change / PR for #2275 was later deemed unfit and closed out...I don't see any other avenues on how this is going to be resolved. Can you please clarify if I missed another PR or open issue?
> 
> The PR in question from #2275 can be found here: #2755

Yeah. I didn't realize it was a dup until after I fixed it. The PR that fixed it was #2797. #2755 was an incorrect approach to solving this problem.

**Comment by carlosmkb** [comment]
the paste problem was not resolved. please check https://github.com/microsoft/terminal/issues/2275#issuecomment-534936850

**Comment by willemdh** [comment]
For me the paste problem is also not resolved :(

**Comment by apenney** [comment]
With the current release when I cut and paste I get one giant line with no linebreaks at all.

**Comment by pkoscisz** [comment]
This is a huge issue, how anybody can be working with this tool, please fix ASAP.....

**Comment by Ge0rges** [comment]
I still get this copying from Ubuntu 18.04 on latest version of WSL (in Windows10) and pasting to stackoverflow/pastebin in Firefox (latest), and NotePad (latest).

**Comment by CosmicCatnap** [comment]
I've been using the Windows Terminal now since it was first pushed, all the way back to when I had to compile each version and this has never worked. Attempting to copy and paste using the shortcuts has issues and the default ctrl+c ctrl+v functionality clearly has problems as well. Combine that with the related issues caused by WSL and I have to question who this is even being built for at this point. 

Its nice that you have CI/CD going but clearly its not got great regression testing because I've seen some variant of this bug open/closed/ and then reopened the whole time with no comprehensive solutions. What I don't understand is how every time I've been able to reproduce it easily within minutes so how is this code getting pushed without being tested? 
it seems to me that this is all hamstrung to issues inherent in ConPTY which hasn't seen a push in 5 months which smells of project priority issues to me and really should have had a solid year of work on it before this ever saw the light of day on github, because pushing a Terminal that can't even copy and paste at half a year in is paramount to saying "hopefully someone else will fix this for me in the open source community" which is a perfectly fine request of a 1.0 > project with promised features delivered but somewhat distasteful at what appears to be a prototyping phase.

For those of us who are using this to SSH into *nix systems and use vim on our windows box at work which I assume is quite a few of us, or are prototyping in WSL even if just for small things...this piece of software is effectively _useless_ until it can at least pretend to have basic parity with a *nix terminal or even the CMD. Even trying to copy and paste in a CMD session between two files in windows is a dubious proposition at best which makes this an inferior product to the default command prompt bundled with the OS currently.

I don't need tabs, I don't need Acrylic, I don't need some cool new font, in fact the one aesthetic thing that would have been nice was an actual settings menu instead of a profile.json which is another ticket that has been open for months in various states of incomplete. 

In summary I need your product to work at a fundamental level where I can actually use it! I don't expect it to be perfect or bug free at any point but its clear you've focused on the wrong things when determining who would want this, what for, and why because currently its just a sad conversation piece about what could have been and talk about how excited and naive we were to think this would turn out well in 6 months.

**Comment by postwait** [comment]
This is a pretty brutal bug.  Just a note to help indicate criticality.

**Comment by alxtsg** [comment]
On Windows 10 Pro (version 1903), using Windows Terminal (version 0.5.2762.0) and PowerShell Core (version 6.2.2), I am having the same problem. I connect to multiple Linux-based systems in my daily job, this issue prevents me from copy-and-paste commands/ text effectively.

**Comment by gohar94** [comment]
Is nobody working on this?

**Comment by DHowett-MSFT** [comment]
At this exact moment, we've been focusing on making sure the in-box console continues to work well for the upcoming release of Windows. Sorry about the radio silence! 😃

**Comment by mikkushukla** [comment]
This is the basic need of any ssh client l, we can take learning from putty how consistently it is working.  Even Android client are good at copy paste.

Get Outlook for Android<https://aka.ms/ghei36>

________________________________

**Comment by EdLy3** [comment]
This problem appeared in a DOS window with my upgrade to 1903.
Is the DOS window code driving this?  I can no longer copy
out of my Hessling editor.  I am ok if I save the file and
use notepad but this is a real nuisance.

**Comment by pkoscisz** [comment]
I hope you are kidding, you have major bug that affect the way people interact with the tool and there is no focus on fixing it but you do have time to upgrading it to new windows versions....

**Comment by mikkushukla** [comment]
Take learning from mac terminal, if you are building a good echo system, you should focus on terminal. Event open-source ubuntu terminal work like charm.

Get Outlook for Android<https://aka.ms/ghei36>

________________________________

**Comment by alxtsg** [comment]
> I hope you are kidding, you have major bug that affect the way people interact with the tool and there is no focus on fixing it but you do have time to upgrading it to new windows versions....

Well technically this software is still in preview stage and not even stable yet...

**Comment by mcornella** [comment]
Everyone relax, this is a beta product, don't use it for production. There are other Microsoft terminal programs that function properly so just use those while this isn't working.

**Comment by pkoscisz** [comment]
> > I hope you are kidding, you have major bug that affect the way people interact with the tool and there is no focus on fixing it but you do have time to upgrading it to new windows versions....
> 
> Well technically this software is still in preview stage and not even stable yet...

This is community's feedback which tries to use the tool, i think the major frustration is that nothing it happening on tool side, especially for major bug as this which was reported months ago...

Would be nice if somebody just let us know that you guys are looking into it.

>  There are other Microsoft terminal 

LOL, which ones???

**Comment by mcornella** [comment]
> LOL, which ones???

- The one provided when installing a WSL diestro (normally Ubuntu)
- Powershell console
- cmd console

**Comment by pkoscisz** [comment]
If you want to stay in ubuntu subsystem, ubuntu one is the only option however as we all know it is basic so we all have been hoping that **The Terminal** could be the tool.

Please prove us right :)

**Comment by awarmfastbear** [comment]
To those who are impatient and contributing nonse like 'Copying and pasting works in Putty, it should work here!', do consider that the developers of this are dealing with higher severity issues crashes, freezing, rendering issues, etc. Those critical issues can rightfully take priority over this. If you want this resolved sooner, you can help debug the issue, apply to their development team or contribute your own code. 

/$0.02

**Comment by mikkushukla** [comment]
Even putty like copy paste is not working in windows terminal, copy multiline text and it will paste your content with extra \n per line which is very irritating, every time i have to delete extra line.

Yes you are right freeze is very common whenever ssh host startresponding slowly.

Get Outlook for Android<https://aka.ms/ghei36>

________________________________

**Comment by d-bingham** [comment]
Looks like this one will be fixed shortly.

**Comment by DHowett-MSFT** [comment]
To give credit where credit is due, _that fix is thanks to @d-bingham_. 😸

**Comment by nerddtvg** [comment]
Thank you, @d-bingham, from all of us waiting on the issue.

**Comment by pkoscisz** [comment]
Now, can we get feature to move tabs around by using cursor and if you have many tabs they don't hide behind '+' character please! LOL 💃

**Comment by DHowett-MSFT** [comment]
It's not on-topic for this particular issue, but the tabs are getting a bit of a boost in 0.6. See the tree of issues linked from #3027.

**Comment by carlosmkb** [comment]
yes, I can confirm that still doesn't work on `v0.6.2951.0`

Copy from VSCode and paste in Windows Terminal on vim:
![image](https://user-images.githubusercontent.com/7363029/67490584-5cc72780-f66b-11e9-84a5-66c9b25e9a55.png)

**Comment by nerddtvg** [comment]
This bug is not related to pasting *into* Terminal, it is pasting *from* Terminal. I believe what you're referring to is discussed in #1091.

**Comment by tirelibirefe** [comment]
I use different terminal emulators in Win10 1903 and all of them have the same problem.

**Clipboard:** govc find / -type m -name 'dev1-*' | xargs govc vm.ip -a -v4
**Pasted as:** govcfind/-typem-name'dev1-*'|xargsgovcv4

**Comment by DHowett-MSFT** [comment]
Those are definitely not line breaks that you’re missing... would you mind filing a new separate issue and following the issue template? Thanks.

**Comment by DHowett-MSFT** [comment]
For what it’s worth: if multiple terminal emulators experience the same issue, it’s probably something in your shell.

**Comment by mikkushukla** [comment]
Terminal freezes during post laptop sleep and network disconnect. Every time I have to force kill the app via task manager.
Is this a known issue?


Get Outlook for Android<https://aka.ms/ghei36>
________________________________

**Comment by nerddtvg** [comment]
@mikkushukla, please open a new issue to report the problem: https://github.com/microsoft/terminal/issues

**Comment by alxtsg** [comment]
@mikkushukla 

> Terminal freezes during post laptop sleep and network disconnect. Every time I have to force kill the app via task manager.
> Is this a known issue?

The problem you described is not related to this issue.

**Comment by pztrn** [comment]
Windows Terminal (Preview)
Version: 0.7.3291.0

Still experiencing this.

**Comment by mikkushukla** [comment]
Because of freeze issue I have stopped using terminal. I am using command prompt as it is more reliable.  If this is fixed in 0.0.3291.0 I will start using again and will update If i face the same issue.


Get Outlook for Android<https://aka.ms/ghei36>

________________________________

**Comment by nerddtvg** [comment]
@pztrn, you may be best served opening anew issue and referencing this one. Ensure you provide all relevant information to reproduce the problem.

@mikkushukla: Your statement has nothing to do with this bug report. If you have issues that are unrelated to line breaks in copy and paste functionality, please open a new bug report.

**Comment by eromoe** [comment]
meet this in terminal 1.0
1. hold ctrl + shift , use cursor to copy multiple lines.
2. paste  in vscode or other editors ,  lose  line break

**Comment by JesusCoyotzi** [comment]
Found this issue too
terminal 1.0 

If I copy from terminal pasting preserves multiple lines. But If I am inside vim when copying and pasting the line breaks dissapear. Resulting in one long line.

My full environment is: Windows terminal, powershell, logged via ssh to a linux box that is running tmux and inside is running vim.

**Comment by carlos-zamora** [comment]
@JesusCoyotzi could you submit a new bug report [here](https://github.com/microsoft/terminal/issues/new?assignees=&labels=&template=Bug_Report.md&title=)? Please include specific steps on how you performed the copy/paste and what you're running there. Sounds like a real issue but we should be tracking this in more detail separately.

**Comment by 89z** [comment]
https://github.com/nsf/termbox-go/issues/237#event-4636645181

**Comment by 89z** [comment]
https://github.com/microsoft/terminal/issues/1073
https://github.com/microsoft/terminal/issues/1126
https://github.com/microsoft/terminal/issues/1621
https://github.com/microsoft/terminal/issues/2275
https://github.com/microsoft/terminal/issues/2278
https://github.com/microsoft/terminal/issues/2324
https://github.com/microsoft/terminal/issues/2395
https://github.com/microsoft/terminal/issues/2717
https://github.com/microsoft/terminal/issues/52

**Comment by zd3nik** [comment]
I am on a new machine using Windows 11 and I'm experiencing this problem in the WSL console.

1. Open a WSL console
2. Do an ls or something to get multiple lines in the console
3. Select/Copy multiple lines
4. Paste into any text editor

The pasted text will be one line with spaces between the text that should be on separate lines.

![image](https://user-images.githubusercontent.com/1753175/211405349-333e66f1-b576-4e0e-985f-83505dae927b.png)
![image](https://user-images.githubusercontent.com/1753175/211405437-5001ff67-0b06-4743-a79e-c07619dc9138.png)

**Comment by zadjii-msft** [comment]
@zd3nik That's by design for <kbd>Ctrl+shift+c</kbd>. See https://github.com/microsoft/terminal/issues/65#issuecomment-437398197

Copy-paste is significantly more reasonable and configurable in the Windows Terminal, I'd recommend that. Conhost's copy-paste is pretty significantly burdened by years of legacy behavior.

**Comment by zd3nik** [comment]
That's unfortunate.  Because automatically copying whatever you select into the clipboard is.... not ideal.  Especially when simply clicking somewhere in the terminal (say, to put focus on the terminal) highlights part of the terminal text (one character).  So I'm constantly blowing whatever useful data I may have had in my clipboard whenever I click on a windows terminal window.  Not ideal.

If the terminal was at least smart enough to only highlight when you "drag" the mouse pointer over part of the terminal instead, then it wouldn't be so detrimental to automatically put whatever you highlight into the clipboard.

**Comment by zd3nik** [comment]
Thanks for the info @zadjii-msft. Is there a way to select which terminal is used for wsl?  I'm just using the terminal that it opens when I launch my wsl console.

**Comment by zadjii-msft** [comment]
Check out [this blog post](https://devblogs.microsoft.com/commandline/windows-terminal-as-your-default-command-line-experience/)

**Comment by picorealm** [comment]
I'm having a problem copying lines of yaml files and then pasting them into Vim in another window.  Windows Terminal is filling the space between the first line and the second line with spaces and fails to copy the newline.  Oddly, it only messes up the first two lines.  Subsequent lines paste correctly.  Here's an example from a .yml file:

- name: Install MariaDB<newline>
  ansible.builtin.apt:<newline>
    pkg:<newline>
    - mariadb-client<newline>
    - mariadb-server<newline.

What gets pasted in the other window is:
- name: Install MariaDB<space><space><space><space><space><space><space><no newline>
  ansible.builtin.apt:<newline>
    pkg:<newline>
    - mariadb-client<newline>
    - mariadb-server<newline>

Options:
  Automatically copy selection to clipboard
  Remove trailing white-space in rectangular selection
  Remove trailing white-space when pasting

Windows Terminal Preview
Version: 1.22.3232.0

**Comment by picorealm** [comment]
Just tested and neither of these settings has any effect on the problem:
  Remove trailing white-space in rectangular selection
  Remove trailing white-space when pasting
